.github/workflows/lint.yml

```diff
@@ -42,27 +48,29 @@ jobs:
   sources-up-to-date:
     name: Check Sources Up-to-Date
     runs-on: ubuntu-22.04
-    container:
-      image: ghcr.io/pulp-platform/snitch_cluster:main
+    if: >
+      github.event_name != 'pull_request' ||
+      github.event.pull_request.head.repo.full_name != github.repository
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: true
-      - name: Install local Python packages
-        run: pip install .
+      - name: Install bender
+        uses: pulp-platform/pulp-actions/bender-install@v2
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+          cache: 'pip'
+      - name: Install pip dependencies
+        run: |
+          pip install .
```

Comment:
The docker build for the branch is in the `ci.yml` workflow, and I wanted to prevent workflow dependencies. But fetching the container from `main` is probably also fine, so we can also revert this.

Reply:
I'm mainly scared of having different dependencies in different places, e.g. the Python version here is already different from the one we use in the container :grin:

Is there a particular reason to avoid workflow dependencies? Or is just more complex?

Reply:
It is just more complex/verbose. But I agree, it would be better to use the docker container. I will add a workflow dependency.

Reply:
I reconsidered 😄 workflow dependencies are annyoing, since they apply to the entire workflow. So with the current workflows, we would have to wait for the entire `ci.yml` workflow to run before the `lint.yml` can start. Also, we would have to [filter out workflow failures](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow), which will then just show up as not run instead of failures.

Long story short, I am not a big fan of workflow dependencies🫠

Reply:
But regarding the first problem we can just move the Docker build to its own workflow as it was, and it is solved.

Regarding the second, I think the dependent job not failing is not a problem, as for it to be reported as not being run the Docker build would already have to fail. Or did I get it wrong?

Reply:
I tried to implement that, but the workflows weren't triggered and  saw that the documentation of [`workflow_run`](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_run) says:

> Note: This event will only trigger a workflow run if the workflow file is on the default branch.

So, I believe workflow dependencies are not an option since this only works on the `main` branch for some reason

Reply:
Oh, well if it's a Github limitation then we might as well put everything in a single workflow file, or leave it as you suggested before.

Reply:
I moved the `sources-up-to-date` job from the lint to the CI workflow, so we can encode the dependency to the docker build job and execute it on the Docker container.